test/merge-patient-lab-iips-mocha.js

```diff
@@ -0,0 +1,138 @@
+// merge-patient-lab-iips-mocha.js
+
+var chai = require('chai');
+var chaiAsPromised = require('chai-as-promised');
+chai.should();
+chai.use(chaiAsPromised);
+
+var _ = require('underscore');
+var noflo = require('noflo');
+var componentFactory = require('../components/merge-patient-lab-iips.js');
+
+describe('merge-patient-lab-iips', function() {
+    var component = createComponent(componentFactory);
+    it("should have rdfpf object on nodeInstance", function() {
+        return Promise.resolve(component).then(function(component){
+            return component.rdfpf;
+        }).should.be.an('object');
+    });
+    it("should have createVni function", function() {
+        return Promise.resolve(component).then(function(component){
+            return component.rdfpf.createVni;
+        }).should.be.a('function');
```

Comment:
I forgot to discuss on today's call: should we make these functions be properties of the various runtime objects, or just functions that are exported from the appropriate module?  For example, call component.rpf.createVni( ...) or call createVni(component, ...)?  If they are properties of the objects then the object uses more memory and requires property chain lookup.  On the other hand, I guess the benefit of being attached to the object is that the functions are bundled with the object.  http://www.toptal.com/javascript/javascript-prototypes-scopes-and-performance-what-you-need-to-know  But I would think that we could still bundle the functions into a module, using module.exports even if they were not attached to the runtime objects.  The memory usage issue would be more applicable to VNI and State objects than anything else.

It sort of seems to be that it would be better to NOT store the functions in the objects, but instead pass the object explicitly as the first parameter, like: createVni(component, ...).  What do you think, and why?

I do not want to do premature optimization here, but I do want to be aware of the trade-offs that we are making, and make the best trade-off.  I also would prefer to do things consistently one way or the other, and for that I think the explicit parameter passing style would probably be more general.  I have also found that it is considerably easier to read code that uses an explicit parameter instead of passing an object implicitly using 'this', because it is not easy to know what 'this' is supposed to be bound to.  Explicit parameters typically have descriptive names.

Some discussion here:
https://news.ycombinator.com/item?id=4985853